psc/src/main/java/com/pinterest/psc/consumer/memq/MemqOffset.java

```diff
@@ -0,0 +1,64 @@
+package com.pinterest.psc.consumer.memq;
+
+
+/* PSC offsets for memq messages:
+ * |-1 bit----|--------- 45 bit ---------|------18 bit ------|
+ * | reserved |          batch           |      message      |
+ * |          |          offset          |      offset       |
+ * |----------|--------------------------|-------------------|
+ *
+ * e.g. 0xffff_e000_0000_1000L -> batch offset 262143, message offset 4096
+ *
+ * Note:
+ * This has to be structured this way since PscConsumerThread will perform increments on the offsets,
+ * which should be equivalent to adding to message offsets
+ */
+public class MemqOffset {
```

Comment:
I think for now we can leave it in PSC codebase, since the `MemqOffset` definition within MemQ itself would require some discussion with Ambud regarding whether introducing this concept would be appropriate. If this class does become obsolete in PSC, we can always remove its definition and rely on the MemQ definition.